### PR TITLE
Fix selection early termination test

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Credit to: siddharthteotia
Replace: #5167

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Replace #5167

File: pinot-core/src/test/java/org/apache/pinot/queries/SelectionOnlyEarlyTerminationTest.java [hunk 1]
```diff
@@ -61,8 +62,13 @@ public void testSelectOnlyQuery() {
...
       assertEquals(brokerResponse.getNumSegmentsProcessed(), numSegmentsPerServer * NUM_SERVERS);
-      assertEquals(brokerResponse.getNumSegmentsMatched(), numThreadsPerServer * NUM_SERVERS);
-      assertEquals(brokerResponse.getNumDocsScanned(), numThreadsPerServer * NUM_SERVERS * limit);
+      // NOTE: 'numSegmentsMatched' and 'numDocsScanned' could be in a range because when the CombineOperator second
+      //       phase merge early terminates, the operators might not finish scanning the documents
+      long numSegmentsMatched = brokerResponse.getNumSegmentsMatched();
+      assertTrue(numSegmentsMatched >= NUM_SERVERS && numSegmentsMatched <= numThreadsPerServer * NUM_SERVERS);
+      long numDocsScanned = brokerResponse.getNumDocsScanned();
+      assertTrue(
+          numDocsScanned >= NUM_SERVERS * limit && numSegmentsMatched <= numThreadsPerServer * NUM_SERVERS * limit);
```
Comment (mcvsubbu) [inline pinot-core/src/test/java/org/apache/pinot/queries/SelectionOnlyEarlyTerminationTest.java]:
```suggestion
          numDocsScanned >= NUM_SERVERS * limit && numDocsScanned <= numThreadsPerServer * NUM_SERVERS * limit);
```

File: pinot-core/src/test/java/org/apache/pinot/queries/SelectionOnlyEarlyTerminationTest.java [hunk 2]
```diff
@@ -72,8 +78,13 @@ public void testSelectOnlyQuery() {
...
       assertEquals(brokerResponse.getNumSegmentsProcessed(), numSegmentsPerServer * NUM_SERVERS);
-      assertEquals(brokerResponse.getNumSegmentsMatched(), numThreadsPerServer * NUM_SERVERS);
-      assertEquals(brokerResponse.getNumDocsScanned(), numThreadsPerServer * NUM_SERVERS * limit);
+      // NOTE: 'numSegmentsMatched' and 'numDocsScanned' could be in a range because when the CombineOperator second
+      //       phase merge early terminates, the operators might not finish scanning the documents
+      numSegmentsMatched = brokerResponse.getNumSegmentsMatched();
+      assertTrue(numSegmentsMatched >= NUM_SERVERS && numSegmentsMatched <= numThreadsPerServer * NUM_SERVERS);
+      numDocsScanned = brokerResponse.getNumDocsScanned();
+      assertTrue(
+          numDocsScanned >= NUM_SERVERS * limit && numSegmentsMatched <= numThreadsPerServer * NUM_SERVERS * limit);
```
Comment (mcvsubbu) [inline pinot-core/src/test/java/org/apache/pinot/queries/SelectionOnlyEarlyTerminationTest.java]:
```suggestion
          numDocsScanned >= NUM_SERVERS * limit && numDocsScanned <= numThreadsPerServer * NUM_SERVERS * limit);
```

**Comment by Jackie-Jiang** [comment]
@mcvsubbu Good catch!

**Pull request merged**